compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -211,9 +201,11 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         }
     }
 
-    pub fn typing_mode(&self) -> TypingMode<'tcx> {
+    /// During CTFE we're always in `PostAnalysis` mode.
+    #[inline(always)]
+    pub fn typing_env(&self) -> ty::TypingEnv<'tcx> {
         debug_assert_eq!(self.param_env.reveal(), Reveal::All);
-        TypingMode::PostAnalysis
+        ty::TypingEnv { typing_mode: ty::TypingMode::PostAnalysis, param_env: self.param_env }
```

Comment:
This is also used by Miri. So if Miri should use TypingMode::fully_monomorphized()`, then something seems wrong here?

Reply:
the `param_env` is always empty in MIRI, so `ty::TypingEnv { typing_mode: ty::TypingMode::PostAnalysis, param_env: self.param_env }`  is equal to `ty::TypingEnv::fully_monomorphized()`

Reply:
Ah, okay.